packages/shared/components/organisms/ProposalCard.svelte

```diff
@@ -9,10 +9,18 @@
     import { Icon } from '@auxiliary/icon/enums'
     import { localize } from '@core/i18n'
     import { proposalsState } from '@contexts/governance/stores'
+    import { isVotingForProposal } from '@contexts/governance/utils'
 
     export let proposal: IProposal
 
+    let hasVoted = false
+
     $: proposalState = $proposalsState[proposal?.id]
+    $: void setHasVoted()
+
+    async function setHasVoted(): Promise<void> {
+        hasVoted = await isVotingForProposal(proposal?.id)
+    }
```

Comment:
This could be inside of an `onMount`:
```typescript
let hasVoted: boolean

async function onMountHelper(): Promise<void> {
    hasVoted = await isVotingForProposal(proposal?.id)
}

onMount(() => void onMountHelper())
```